project/Settings.scala

```diff
@@ -178,11 +178,91 @@ object Settings {
     )
   }
 
+  lazy val scripted100M6Workaround = {
+
+    // see https://github.com/sbt/sbt/issues/3245#issuecomment-306045952
+
+    ScriptedPlugin.scripted := Def.inputTask {
+
+      val args = ScriptedPlugin
+        .asInstanceOf[{
+          def scriptedParser(f: File): complete.Parser[Seq[String]]
+        }]
+        .scriptedParser(sbtTestDirectory.value)
+        .parsed
+
+      val prereq: Unit = scriptedDependencies.value
+      val scriptedTests = ScriptedPlugin.scriptedTests.value
+
+      val testDir = sbtTestDirectory.value
+      val log = scriptedBufferLog.value
+      val args0 = args.toArray
+      val launcher = sbtLauncher.value
+      val opts = scriptedLaunchOpts.value.toArray
+
+      val sbtv = sbtVersion.in(pluginCrossBuild).value
+
+      try {
+        if(sbtv == "1.0.0-M6")
+          scriptedTests.asInstanceOf[{
+            def run(
+              x1: File,
+              x2: Boolean,
+              x3: Array[String],
+              x4: File,
+              x5: Array[String],
+              x6: java.util.List[File]
+            ): Unit
+          }].run(testDir, log, args0, launcher, opts, new java.util.ArrayList)
+        else
+          scriptedTests.asInstanceOf[{
+            def run(
+              x1: File,
+              x2: Boolean,
+              x3: Array[String],
+              x4: File,
+              x5: Array[String]
+            ): Unit
+          }].run(testDir, log, args0, launcher, opts)
+      } catch {
+        case e: java.lang.reflect.InvocationTargetException =>
+          throw e.getCause
+      }
+    }.evaluated
+  }
+
   lazy val plugin =
     javaScalaPluginShared ++
     divertThingsPlugin ++
     withScriptedTests ++
     Seq(
+      sbtLauncher := {
```

Comment:
The [default `sbtLauncher` implem](https://github.com/sbt/sbt/blob/6ce09cfed6f3353326981a4e6e07ffba1f15252f/scripted/plugin/src/main/scala/sbt/ScriptedPlugin.scala#L102) expects the sbt launcher to appear first in the update report.  But coursier currently gives no guaranties about the order of the modules and artifacts in its output, so not in the update report either. For whatever reasons, it was in first position anyway before, but isn't anymore now...